File: bukkit/bukkit-cmd/src/main/java/me/raider/plib/bukkit/cmd/supplier/PlayerSupplier.java
package me.raider.plib.bukkit.cmd.supplier;

import me.raider.plib.commons.cmd.ArgumentSupplier;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class PlayerSupplier implements ArgumentSupplier<Player> {

    @Override
    public Player supply(Object object) {
        if (object instanceof Player) {
            return (Player) object;
        }
        if (object instanceof String) {
            return Bukkit.getPlayer((String) object);
        }

        return null;
    }
}

File: bukkit/bukkit-cmd/src/main/kotlin/me/raider/plib/bukkit/cmd/suppliers/PlayerSupplier.kt
package me.raider.plib.bukkit.cmd.suppliers

import me.raider.plib.commons.cmd.ArgumentSupplier
import org.bukkit.Bukkit
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class PlayerSupplier : ArgumentSupplier<Player> {

    override fun supply(`object`: Any?): Player? {
        if (`object` is CommandSender) {
            if (`object` is Player) {
                return `object`
            }
            return null
        }
        if (`object` == null) {
            return null
        }

        return Bukkit.getPlayer(`object` as String)
    }
}